lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -744,15 +744,25 @@ instance ToText FeePolicy where
 
 instance FromText FeePolicy where
     fromText txt = case T.splitOn " + " txt of
-        [a, b, c] | T.takeEnd 1 b == "x" && T.takeEnd 1 c == "y"
+        [a, b, c]
+            |  T.takeEnd 1 b == "x"
+            && T.takeEnd 1 c == "y"
+            && all isDouble [a, T.dropEnd 1 b, T.dropEnd 1 c]
                     -> LinearFee
                    <$> fmap Quantity (fromText a)
                    <*> fmap Quantity (fromText (T.dropEnd 1 b))
                    <*> fmap Quantity (fromText (T.dropEnd 1 c))
         _ ->
             Left $ TextDecodingError
                 "Unable to decode FeePolicy: \
-                \Linear equation not in expected format: a + bx + cy"
+                \Linear equation not in expected format: a + bx + cy \
+                \or a, b, c are not numbers"
+     where
+         isDouble :: Text -> Bool
+         isDouble a = case (fromText @Double a) of
+             Right _ -> True
+             Left _ -> False
```

Comment:
**[suggestion: alternative]**

Leverage `left` (works like `fmap`, but on the left part of a `Either`) from `Control.Arrow`

```hs
import Control.Arrow
    ( left )

{- ... -}

instance FromText FeePolicy where
    fromText txt = case T.splitOn " + " txt of
        [a, b, c] | T.takeEnd 1 b == "x" && T.takeEnd 1 c == "y" ->
            left (const err) $ LinearFee
                <$> fmap Quantity (fromText a)
                <*> fmap Quantity (fromText (T.dropEnd 1 b))
                <*> fmap Quantity (fromText (T.dropEnd 1 c))
        _ ->
            Left err
      where
        err = TextDecodingError
            "Unable to decode FeePolicy: \
            \Linear equation not in expected format: a + bx + cy \
            \where 'a', 'b', and 'c' are numbers"
```

**Why?**

The solution is slightly more concise and readable IMO, and also, a bit more general in terms or error handling.

Reply:
Fixed as suggested.
Although the definition of `left` in `Control.Arrow` is very obscure to me... :sweat_smile: 
```haskell
left :: a b c -> a (Either b d) (Either c d)
```

Reply:
Because this definition is missing a crucial thing :p 

`a` is an `Arrow` (and even an `ArrowChoice`). 

So it's actually: 

```hs
left :: ArrowChoice a => a b c -> a (Either b d) (Either c d)
```

Now, if you think of a as an actual arrow, and replace `a` with an arrow and use it as an infix operator, it becomes less cryptic :) :

```hs
type a = (->) -- This is just for the sake of understanding

left :: (b -> c) -> (Either b d -> Either c d)
```